starknet_py/net/account/account_deployment_result.py

```diff
@@ -0,0 +1,14 @@
+from dataclasses import dataclass
+
+from starknet_py.contract import SentTransaction
+from starknet_py.utils.sync import add_sync_methods
+
+
+@add_sync_methods
+@dataclass(frozen=True)
+class AccountDeploymentResult(SentTransaction):
+    account: "Account" = None  # pyright: ignore
```

Comment:
Does it behave nicely in documentation? I think it was problematic there. 
We can do this instead: https://stackoverflow.com/a/39757388 (__future__.annotations, import based on TYPE_CHECKING).

Reply:
I think it's not really documented i.e there's no section for `AccountDeploymentResult`. Should that be changed?

Reply:
Added a section in `account.rst`. It looks like this 
![image](https://user-images.githubusercontent.com/52135326/207304516-dca6dd26-d769-4413-b267-682cdcaabfbf.png)

It's basically the same as current docs of DeployResult and DeclareResult look in Contract section.

Reply:
Hmm I think we should include properties that we want to expose.